localstack-core/localstack/aws/handlers/region.py

```diff
@@ -15,9 +21,83 @@ class RegionContextEnricher(Handler):
 
     def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
         context.region = self.get_region(context.request)
+        context.partition = get_partition(context.region)
 
     @staticmethod
     def get_region(request: Request) -> str:
         from localstack.utils.aws.request_context import extract_region_from_headers
 
         return extract_region_from_headers(request.headers)
+
+
+class RegionRewriterStrategy(abc.ABC):
+    @abc.abstractmethod
+    def apply(self, context: RequestContext):
+        """
+        Apply the region rewriter to the request context
+        :param context: Request Context
+        """
+        pass
+
+
+class DefaultRegionRewriterStrategy(RegionRewriterStrategy):
+    """
+    If a region is not known, override it to "us-east-1"
+    """
+
+    default_region = "us-east-1"
+
+    def apply(self, context: RequestContext):
+        if not context.region:
+            return
+
+        if context.region not in self.available_regions:
+            LOG.warning(
+                "Region '%s' is not available. Resetting the region to 'us-east-1'. "
+                "Please consider using a region in the 'aws' partition to avoid any unexpected behavior. "
+                "Available regions: %s",
+                context.region,
+                self.available_regions,
+            )
+            context.region = self.default_region
+            context.partition = "aws"
+            self.rewrite_auth_header(context, self.default_region)
+
+    def rewrite_auth_header(self, context: RequestContext, region: str):
+        """
+        Rewrites the `Authorization` header to reflect the specified region.
+        :param context: Request context
+        :param region: Region to rewrite the `Authorization` header to.
+        """
+        auth_header = context.request.headers.get("Authorization")
+
+        if auth_header:
+            regex = r"Credential=([^/]+)/([^/]+)/([^/]+)/"
+            auth_header = re.sub(regex, rf"Credential=\1/\2/{region}/", auth_header)
+            context.request.headers["Authorization"] = auth_header
+
+    @cached_property
+    def available_regions(self) -> list[str]:
+        """
+        Returns a list of supported regions.
+        :return: List of regions in the `aws` partition.
+        """
+        # We cannot cache the session here, as it is not thread safe. As the entire method is cached, this should not
+        # have a significant impact.
+        # using s3 as "everywhere available" service, as it usually is supported in all regions
+        # the S3 image also deletes other botocore specifications, so it is the easiest possibility
+        return Session().get_available_regions("s3", "aws")
```

Comment:
nit: IAM or STS might be an even better candidate, but I don't think it will make any practical difference here

Reply:
Yes, I initially had `ec2` here, but changed it for `s3`. The main issue is the `s3` image, where we delete all botocore specs except those of s3, so the regions will be empty for all other services. To avoid removing this deletion, I used s3, as it is also in all regions. (I hope this is clear from the comments)